Documentation/storage.md

```diff
@@ -0,0 +1,161 @@
+# Storage options
+
+Dex requires persisting state to perform various tasks such as track refresh tokens, preventing replays, and rotating keys. This document is a summary of the storage configurations supported by dex.
+
+Regardless of the implementation, dex saves sensitive data in its backing storage: signing keys, bcrypt'd passwords, etc. Storage breaches are serious because attacks may propagate to any application which relies on dex. Whatever storage option used, transport security and database ACLs are both musts.
+
+## Caveat: Running replicated instances
+
+Validation is still being done to ensure correctness when multiple instances of dex are using the same backing storage. While everything "should just work", edge cases require time to investigate and write tests for. This work can be followed on the issue tracker [here](https://github.com/coreos/dex/issues/600).
```

Comment:
> Dex performs correctness validation when multiple instances use the same backing storage. 

That's not the same thing. This is stating that we still need to write the tests to prove correctness and you probably shouldn't be running these kind of setups in the first place. How does this sound?

> Tests still need to be written to validate that multiple instances of dex behave correctly when using the same storage.
> While there aren't any technical limitations, edge cases have been observed and progress on these kind of bugs can be found on the `[dex issue tracker]`.